src/services/offer.js

```diff
@@ -212,44 +212,10 @@ class OfferService {
             score: lastOfferScore
         } = queryToken ? this.decodeQueryToken(queryToken) : {};
 
-        let offers;
-        if (lastOfferId && value) {
-            offers = Offer.aggregate([
-                { $match: { $text: { $search: value } } },
-                { $addFields: {
-                    score: { $meta: "textScore" },
-                    adminReason: { $cond: [showAdminReason, "$adminReason", "$$REMOVE"] }
-                } },
-                { $match: { "$or": [
-                    { score: { "$lt": lastOfferScore } },
-                    { score: lastOfferScore, _id: { "$gt": ObjectId(lastOfferId) } }
-                ] } },
-                { $match: Offer.filterCurrent() },
-                { $match: showHidden ? {} : Offer.filterNonHidden() }
-            ]);
-
-        } else {
-            if (lastOfferId) {
-
-                offers = Offer.find({ "$and": [
-                    this._buildFilterQuery(filters),
-                    { _id: { "$gt": ObjectId(lastOfferId) } }
-                ] });
-
-            } else {
-
-                offers = (value ? Offer.find({ "$and": [
-                    this._buildFilterQuery(filters),
-                    { "$text": { "$search": value } }
-                ] }, { score: { "$meta": "textScore" } }
-
-                ) : Offer.find(this._buildFilterQuery(filters)));
-            }
-
-            offers.current();
-            if (!showHidden) offers.withoutHidden();
-            if (!showAdminReason) offers.select("-adminReason");
-        }
+        const offers = lastOfferId && value ?
```

Comment:
Thinking about this, shouldn't the query token include all the search information (i.e. value and filters)? I don't think we should require the client to send the value again if they are already sending a query token to "continue the last search", whatever it is, and which includes certain filters. What if the filters changed now? It does not make sense to treat it as a follow-up search, as the score we use to get next results will have a different meaning.

Reply:
Why would the filters change? If they changed, then that has to be a different search. Including the filters in the token feels redundant to me and doesn't really help the FE, since they already saving the filter information and passing it to the endpoint. We'd need to refactor BE and FE.

But what do you think, @TiagooGomess, as for the FE implementation?

Reply:
You are developing an API (which may have multiple clients and you may not control all of them). It does not make sense to continue a search and change the filters, therefore the token should contain all the information to continue a given search. 

This does not affect the FE, it will just send back what you give it. Even if it sends some filters, whenever there's a queryToken they should be ignored (same for search value)

If anything, this could even simplify the FE, but no changes are really required there

Reply:
@imnotteixeira I implemented this, please review